src/features/projectsV2/ProjectSnippet/microComponents/ImageSection.tsx

```diff
@@ -34,22 +36,31 @@ const ImageSection = (props: ImageSectionProps) => {
   const { embed, callbackUrl } = useContext(ParamsContext);
   const isEmbed = embed === 'true';
 
-  const handleBackButton = useCallback((e: MouseEvent) => {
-    e.stopPropagation();
-    if (document.referrer) {
-      window.history.go(-2);
-    } else {
-      router.replace({
-        pathname: `/${locale}/prd`,
-        query: {
-          ...(isEmbed ? { embed: 'true' } : {}),
-          ...(isEmbed && callbackUrl !== undefined
-            ? { callback: callbackUrl }
-            : {}),
-        },
+  const handleBackButton = () => {
+    if (setPreventShallowPush) setPreventShallowPush(true);
+    setSelectedSite(null);
+    const previousPageRoute = sessionStorage.getItem('backNavigationUrl');
+    const defaultRoute = `/${locale}/prd`;
+    const queryParams = {
+      ...(isEmbed ? { embed: 'true' } : {}),
+      ...(isEmbed && callbackUrl !== undefined
+        ? { callback: callbackUrl }
+        : {}),
+    };
+    const routerPath = previousPageRoute
+      ? previousPageRoute.split('?')[0]
+      : defaultRoute;
+
+    router
+      .push({
+        pathname: routerPath,
+        query: queryParams,
+      })
+      .then(() => sessionStorage.removeItem('backNavigationUrl'))
+      .catch((error) => {
+        console.error('Navigation failed:', error);
       });
-    }
-  }, []);
+  };
```

Comment:
A small issue with this logic. 

When an absolute URL is provided, the embed param / callbackUrl param gets added to the absolute URL after clicking back. So if I had added "https://www.google.com/" as the `backNavigationUrl`